shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -190,6 +195,145 @@ - (void)testStartKeyboardAnimationWillInvokeSetupKeyboardAnimationVsyncClient {
+  FlutterViewController* viewControllerMock = OCMPartialMock(viewController);
+
+  BOOL isLocal = YES;
+  CGFloat screenHeight = UIScreen.mainScreen.bounds.size.height;
+
+  // Start show keyboard animation.
+  CGRect initialShowKeyboardBeginFrame = CGRectMake(0, screenHeight, 0, 0);
+  CGRect initialShowKeyboardEndFrame = CGRectMake(0, screenHeight - 250, 0, 0);
+  NSNotification* fakeNotification = [NSNotification
+      notificationWithName:UIKeyboardWillHideNotification
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameBeginUserInfoKey" : @(initialShowKeyboardBeginFrame),
+                    @"UIKeyboardFrameEndUserInfoKey" : @(initialShowKeyboardEndFrame),
+                    @"UIKeyboardAnimationDurationUserInfoKey" : @(0.25),
+                    @"UIKeyboardIsLocalUserInfoKey" : @(isLocal)
+                  }];
+  viewControllerMock.targetViewInsetBottom = 100;
+  [viewControllerMock handleKeyboardNotification:fakeNotification];
+  BOOL isShowingAnimation1 = viewControllerMock.keyboardAnimationIsShowing;
+  XCTAssertTrue(isShowingAnimation1);
+
+  // Start compounding show keyboard animation.
+  CGRect compoundingShowKeyboardBeginFrame = CGRectMake(0, screenHeight - 250, 0, 0);
+  CGRect compoundingShowKeyboardEndFrame = CGRectMake(0, screenHeight - 500, 0, 0);
+  fakeNotification = [NSNotification
+      notificationWithName:UIKeyboardWillHideNotification
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameBeginUserInfoKey" : @(compoundingShowKeyboardBeginFrame),
+                    @"UIKeyboardFrameEndUserInfoKey" : @(compoundingShowKeyboardEndFrame),
+                    @"UIKeyboardAnimationDurationUserInfoKey" : @(0.25),
+                    @"UIKeyboardIsLocalUserInfoKey" : @(isLocal)
+                  }];
+
+  viewControllerMock.targetViewInsetBottom = 200;
+  [viewControllerMock handleKeyboardNotification:fakeNotification];
+  BOOL isShowingAnimation2 = viewControllerMock.keyboardAnimationIsShowing;
+  XCTAssertTrue(isShowingAnimation2);
+  XCTAssertTrue(isShowingAnimation1 == isShowingAnimation2);
+
+  // Start hide keyboard animation.
+  CGRect initialHideKeyboardBeginFrame = CGRectMake(0, screenHeight - 500, 0, 0);
+  CGRect initialHideKeyboardEndFrame = CGRectMake(0, screenHeight - 250, 0, 0);
+  fakeNotification = [NSNotification
+      notificationWithName:UIKeyboardWillHideNotification
+                    object:nil
+                  userInfo:@{
+                    @"UIKeyboardFrameBeginUserInfoKey" : @(initialHideKeyboardBeginFrame),
+                    @"UIKeyboardFrameEndUserInfoKey" : @(initialHideKeyboardEndFrame),
+                    @"UIKeyboardAnimationDurationUserInfoKey" : @(0.25),
+                    @"UIKeyboardIsLocalUserInfoKey" : @(isLocal)
+                  }];
+
+  viewControllerMock.targetViewInsetBottom = 100;
```

Comment:
By updating the notification names and mocking the `mainScreenIfViewLoaded` and the `viewIfLoaded ` as directed in comments above, you won't need to do this. The `targetViewInsetBottom` should be changed by the notification above.